### ISSUE TITLE
Error values with -k kernel_name option 

### ISSUE THREAD

**Issue by gmarkomanolis** [AMDResearch/omniperf#80]
Using dev branch version, commit 6c1379fccb8c

I have the following issues:

- When running --roof-only with -k flag, then although AI could be almost ok, the BW or FLOPs, could be too high, above the peak, mainly for example with kernel BW above LDS. Also, kernel with more FLOPs than other roofline models from other vendors.

- When visualizing FP32/64 and FP8/16, although I am waiting to have empty FP8/16, then there are data points same AI with FP32/64 but different y values, this seems like a bug.

**Comment by gmarkomanolis** [comment]
I was informed of another case from a colleague using production omniperf (removed real names):

omniperf profile -k kernel_name-n workload -- ./binary -m 0

L2 - Fabric Read BW 342% of the peak
L2 - Fabric Write BW 288% of the peak

**Comment by gmarkomanolis** [comment]
For the second comment above, removing the -k kernel_name solves the issue, so the -k functionality creates the problem.

**Comment by coleramos425** [comment]
I ran an experiment on the workload in question (removed real name). 
![MicrosoftTeams-image](https://user-images.githubusercontent.com/11466906/217859896-fab62eb6-f050-4406-8cff-d1e5bc56e834.png)
On the left you'll see the AI results with kernel filtering as it was applied by rocprofiler. These were the coordinates that exceeded peak LDS BW. 

The right shows AI results I've generated by manually filtering your kernel from a profiling session with no filtering. Significantly lower and in line with our expected result

So the issue you face when using `-k/--kernels` filter isn't occurring in roofline arithmetic, rather at the profiling level. I know this workload is ~6M dispatch so I'm speculating as to whether this has anything to do with the issue. Especially since I couldn't reproduce it with my smaller toy applications...
